pkg/controller/volume/persistentvolume/pv_controller.go

```diff
@@ -1473,6 +1473,15 @@ func (ctrl *PersistentVolumeController) provisionClaimOperation(
 	// called from provisionClaim(), in this case, plugin MUST NOT be nil
 	// NOTE: checks on plugin/storageClass has been saved
 	pluginName := plugin.GetPluginName()
+	if pluginName != "kubernetes.io/csi" && claim.Spec.DataSource != nil {
```

Comment:
Can you please add an unit test somewhere in provision_test.go? Check e.g. how is `claimWithAnnotation()` used there and create `claimWithVolumeSource()` based on it. The rest should be trivial.

Reply:
Done.